src/proc.cpp

```diff
@@ -473,10 +473,9 @@ static bool process_clean_after_marking(bool allow_interactive) {
     // don't try to print in that case (#3222)
     const bool interactive = allow_interactive && cur_term != NULL;
 
-    bool erased = false;
+    static std::vector<job_t *> erase_list;
```

Comment:
This shouldn't be static - we want to move to a multithreaded model with multiple job lists so we should avoid introducing new shared variables. It looks like this can be a local variable.

Also it should store `shared_ptr<job_t>` instead of raw pointers.

Reply:
I wanted to avoid heap allocation each time. Perhaps thread local instead?

Reply:
Also the storing of raw pointers was intentional as the assumption is that lifetime is guaranteed as if our code doesn't run then the shared pointers stay in the list (I.e. references remain), storing the raw pointer prevents the needless increment and decrement of the reference count. 

I always prefer maintainable over performant up until the point that it actually makes a difference. This is a hotpath, I honestly feel the previous iteration with the erase in the for loop was a much better compromise. We can make the increment stage a lambda if you like.